### PR TITLE
DBZ-3785 Bump Quarkus from 2.0.0.Final to 2.2.3.Final

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-3785

### CODE REVIEW

**Comment by Naros** [comment]
So Sanne has reached out with a suggestion that we should move where we hand off the XML mapping to Hibernate ORM from the runtime project to the deployment project.  I have one open question for him so hopefully, I'll get an answer on Friday so we can have this in place for next week's release with the optimal change rather than explicitly adding Jandex ourselves.

**Comment by Naros** [comment]
So after discussions and some trial and error with Sanne, we've concluded that it does not seem technically possible at the moment via build steps to provide an on-the-fly resource that can be consumed in a later build step by ORM so for now the current solution of adding the Jandex artifact is the only viable option.  

In the meantime, I'm working on a PR for Quarkus 2.4 where we'll be able to pass a constructed JaxbHbmMapping instance directly to ORM so we can avoid both the XML encoding/serialization and so that we can provide the instance during the deployment build step phase so that the Jandex dependency can be removed.

**Comment by Naros** [comment]
@jpechane It looks like the test failure for debezium-server might be related, though I haven't had a chance to look into it yet.  Any ideas on why that test could be failing that you can think of?

**Comment by ismailsimsek** [comment]
could be related to https://github.com/quarkusio/quarkus/issues/20190

**Review by gunnarmorling — COMMENTED** [review]
Some comments inline; seems we're blocked here, @Naros?

**Comment by Naros** [comment]
@ismailsimsek I ran each of the modules and the following fail with the same problem:

- Kafka
- Pulsar
- Redis

It's strange these are all built similarly and the above are the only ones that fail with that error.  I also did notice that the NATS implementation has a dependency regression with the upgrade that I'm looking into.

File: debezium-quarkus-outbox/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ metadata:
   - "debezium"
   - "outbox"
   - "cdc"
-  guide: "https://debezium.io/documentation/reference/1.2/integrations/outbox.html"
+  guide: "https://debezium.io/documentation/reference/1.7/transformations/outbox-event-router.html"
```
[the remark this answers is not in the document]
Comment (Naros) [reply debezium-quarkus-outbox/runtime/src/main/resources/META-INF/quarkus-extension.yaml:9]: We should be able to, yes.

File: pom.xml [hunk 2]
```diff
@@ -109,7 +109,7 @@
         <version.antlr4test.plugin>1.18</version.antlr4test.plugin>
 
         <!-- Quarkus -->
-        <quarkus.version>2.0.0.Final</quarkus.version>
+        <quarkus.version>2.2.0.Final</quarkus.version>
```
[the remark this answers is not in the document]
Comment (Naros) [reply pom.xml:112]: It is, I can certainly bump up to that once we work through all the upgrade regressions.

**Comment by ismailsimsek** [comment]
> @ismailsimsek I ran each of the modules and the following fail with the same problem:
> 
>     * Kafka
> 
>     * Pulsar
> 
>     * Redis
> 
>     * NATs
> 
> 
> It's strange these are all built similarly and the above are the only ones that fail with that error.

is it because of skipITs=True? that's one  difference i see in the pom. comparing to kinesis, eventhubs

**Comment by Naros** [comment]
> is it because of skipITs=True? that's one difference i see in the pom. comparing to kinesis, eventhubs

I doubt that @ismailsimsek because the Prevaga adapter POM does not include this explicit property setting and it builds fine.

File: debezium-quarkus-outbox/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gunnarmorling) [reply debezium-quarkus-outbox/runtime/src/main/resources/META-INF/quarkus-extension.yaml:9]: Ok, so are you going to push a commit for this then :)

**Comment by gunnarmorling** [comment]
@Naros, we can keep this for 1.8, right?

**Comment by gunnarmorling** [comment]
If so, can you label it accordingly, please?

**Comment by Naros** [comment]
So @jpechane the associated Quarkus issue, https://github.com/quarkusio/quarkus/issues/20190, suggests adding to the maven failsafe plugin configuration `<useModulePath>false</useModulePath>` and that would resolve that problem; but adding this configuration for example to the redis pom.xml like the following didn't help, any ideas:

```
<plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-failsafe-plugin</artifactId>
                <executions>
                    <execution>
                        <id>integration-test</id>
                        <goals>
                            <goal>integration-test</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>verify</id>
                        <goals>
                            <goal>verify</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <skipTests>${skipITs}</skipTests>
                    <enableAssertions>true</enableAssertions>
                    <useModulePath>false</useModulePath>
                    <systemProperties>
                        <test.type>IT</test.type>
                    </systemProperties>
                </configuration>
            </plugin>
```

**Comment by Naros** [comment]
Ok so I've left a comment on the Quarkus issue because after digging into the `debezium-server` source, the issue is that property resolution has changed with 2.1+ of Quarkus compared to what we experienced with Quarkus 2.0.  Both configuration sources, the one defined in the `debezium-server` artifact and `debezium-server-xxxxxx` are present; however when `ConfigProvider.getConfig()` returns the `Config` object, the later versions seem to give precedence to the configuration values found from the `debezium-server` artifact rather than the `debezium-server-xxxx` artifact.

This is what is leading to the issue of not being able to find a bean named `test` failures because it should be looking for beans named `kafka` for example rather than `test` when running the `debezium-server-kafka` tests.

**Comment by gunnarmorling** [comment]
That's still WIP, right?

**Comment by Naros** [comment]
@gunnarmorling Yes, just discovered the fix discussing with the Quarkus guys so all that remains is your request for the resource filtering to be added to the outbox extension.

**Comment by gunnarmorling** [comment]
@Naros, that build failure looks pretty related, right?

**Comment by Naros** [comment]
@gunnarmorling Yes these are related, I'm looking into it.

**Comment by gunnarmorling** [comment]
Build failure is unrelated and should be handled via https://github.com/debezium/debezium/pull/2759.

**Comment by gunnarmorling** [comment]
Applied; thx, @Naros!

**Pull request merged**